hooks/tk-multi-publish2/maya/publish_files_integ.py

```diff
@@ -0,0 +1,266 @@
+                              }
+                              )
+            return False
+        return self._extra_nodes(nodes)
+
+    def _track_geo_child_naming(self, track_geo):
+        """Checks if the name of nodes under TRACK_GEO are prefixed with 'integ_'.
+            :param:
+                track_geo: nodes under TRACK_GEO
+        """
+        # Nodes under TRACK_GEO group
+        children = cmds.listRelatives(track_geo, c=True)
+        error_names = ""
+        # if there are nodes under TRACK_GEO, check for one without prefix "integ_"
+        if children:
+            for child in children:
+                # If the name doesn't start with integ_ add node name to errorNames
+                if child[:6] != "integ_":
+                    error_names += "\n" + child
+        if error_names:
+            self.logger.error("Incorrect Naming! Node name should start with integ_.",
+                              extra={
+                                  "action_show_more_info": {
+                                      "label": "Show Info",
+                                      "tooltip": "Show the node with incorrect naming",
+                                      "text": "Nodes with incorrect naming:\n{}".format(error_names)
+                                  }
+                              }
+                              )
+            return False
+
+        if children:
+            # if there are nodes under TRACK_GEO, check for locked channels
+            return self._locked_channels(children)
+        else:
+            return True
+
+    def _check_hierarchy(self, group_nodes):
+        """Checks the hierarchy of group nodes in the scene.
+            :param:
+                group_nodes: the list of nodes in the scene
+        """
+        for name in range(len(group_nodes) - 1):
+            # Listing children of group nodes
+            children = cmds.listRelatives(group_nodes[name], c=True)
+            # group_nodes is arranged in hierarchial order i.e. the next node should be the child of previous
+            if children and (group_nodes[name + 1] in children):
+                if name == 'SET_TO_WORLD' and 'CAM' in children:
+                    continue
+            else:
+                hierarchy = "WORLDSCALE\n|__SET_TO_WORLD\n" + "    " + "|__TRACK_GEO\n" + "    " + "|__CAM"
+                self.logger.error("Incorrect hierarchy.",
+                                  extra={
+                                      "action_show_more_info": {
+                                          "label": "Show Info",
+                                          "tooltip": "Show the required hierarchy",
+                                          "text": "Required hierarchy:\n\n{}".format(hierarchy)
+                                      }
+                                  }
+                                  )
+                return False
+        return self._track_geo_child_naming(group_nodes[-1])
```

Comment:
Shouldn't this be getting called from the top level validation() method?

Reply:
The first validation is to check the group node naming and rest of the validations apart from the ones for camera i believe should be depended on that. Calling them from top level validation() method will error out if the node names in the scene is not matching with the global constants.
Currently I'm calling _track_geo_child_naming() after checking node hierarchy but I guess I can call it right after checking node naming.